patsy/categorical.py

```diff
@@ -46,20 +46,37 @@
                         pandas_Categorical_categories,
                         pandas_Categorical_codes,
                         safe_issubdtype,
-                        no_pickling, assert_no_pickling)
+                        no_pickling, assert_no_pickling, check_pickle_version)
+from patsy.state import StatefulTransform
 
 if have_pandas:
     import pandas
 
 # Objects of this type will always be treated as categorical, with the
 # specified levels and contrast (if given).
+
 class _CategoricalBox(object):
     def __init__(self, data, contrast, levels):
         self.data = data
         self.contrast = contrast
         self.levels = levels
 
-    __getstate__ = no_pickling
+    def __getstate__(self):
+        data = getattr(self, 'data')
+        contrast = getattr(self, 'contrast')
+        levels = getattr(self, 'levels')
+        return (0, data, contrast, levels)
+
+    def __setstate__(self, pickle):
+        version, data, contrast, levels = pickle
+        check_pickle_version(version, 0, name=self.__class__.__name__)
```

Comment:
You have to check the pickle version before unpacking it – e.g. if we ever added a 5th field to the pickle state here, then this would error out on the unpacking line, before getting to the `check_pickle_version` line. Basically we have to commit to never change anything that happens before we look at the version, so the first line should be something like
```python
check_pickle_version(pickle[0], 0, ...)   # if we commit to using a (version, ...) tuple forever
```
or
```python
check_pickle_version(pickle["version"], 0, ...)
```
if we decide maybe we want to be stuck with something more readable than a tuple ;-)